src/charm.py

```diff
@@ -343,16 +342,20 @@ def configure_profile(self, profile_name):
                 with attempt:
                     self.k8s_resource_handler.lightkube_client.get(Namespace, name=profile_name)
         except RetryError:
-            self.log.error(f"Action failed. namespace {profile_name} was not created")
+            event.fail(f"Action failed. namespace {profile_name} was not created")
```

Comment:
is this an error saying "was not created" or "does not exist"?  "was not created" suggests to me we are trying to created it but couldn't, but I think here we're looking for an existing namespace?  

I'm not 100% sure what this error is trying to report.  Can you elaborate?

Reply:
Second thought on this: have you tested and seen this error be caught?  I thought that `RetryError` would only be raised if `reraise=False`, and that if `reraise=True` the exception type here would be whatever `lightkube_client.get()` was raising

Reply:
the purpose of this is to wait until the namespace is created. I have not observed it, you've got a point that `reraise` shouldn't be `True` if I'm catching `RetryError`.  I'll test it and adjust.

Reply:
Regarding `reraise=True`, probably that's a good idea (it leads to more descriptive errors being surfaced), just make sure you're catching the original error type and it'll work how you want